ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java

```diff
@@ -770,4 +813,97 @@ private static void verifyPullQueryRow(final Row row) {
     }
     return expectedRows;
   }
+
+  private static Matcher<? super StreamInfo> streamForProvider(
+      final TestDataProvider<?> testDataProvider
+  ) {
+    return streamInfo(testDataProvider.kstreamName(), testDataProvider.topicName(), "JSON");
+  }
+
+  private static Matcher<? super StreamInfo> streamInfo(
+      final String streamName, final String topicName, final String format
+  ) {
+    return new TypeSafeDiagnosingMatcher<StreamInfo>() {
+      @Override
+      protected boolean matchesSafely(
```

Comment:
would we need this if we implemented `.equals()`?

Reply:
I think so, for two reasons:
- I think it's useful for the test to validate the StreamInfo methods (getName, getTopic, getFormat) explicitly, since they're external APIs, rather than simply perform a comparison.
- StreamInfoImpl is package-private so the test can't even create StreamInfo instances for comparison, without exposing StreamInfoImpl which doesn't seem great.

Reply:
ah I had missed the distinction between the impl and the interface